pkg/auth/service.go

```diff
@@ -1012,37 +1092,13 @@ func (s *KVAuthService) ClaimTokenIDOnce(ctx context.Context, tokenID string, ex
 
 // markTokenSingleUse returns true if token is valid for single use
 func (s *KVAuthService) markTokenSingleUse(ctx context.Context, tokenID string, tokenExpiresAt time.Time) (bool, error) {
-	tokenPath := model.KvTokenPath(tokenID)
+	tokenPath := model.KVTokenPath(tokenID)
 	m := model.TokenData{TokenId: tokenID, ExpiredAt: timestamppb.New(tokenExpiresAt)}
-	err := s.store.SetMsg(ctx, model.PartitionKey, tokenPath, &m)
+	err := s.store.SetMsgIf(ctx, model.PartitionKey, tokenPath, &m, nil)
 	if err != nil {
 		return false, err
 	}
```

Comment:
See the DB behaviour here. If it failed do to a predicate - we should return `false, nil`. If it didn't fail, return `true, nil`.

Reply:
return false err

https://github.com/treeverse/lakeFS/blob/2024646a5460f40a337593bf34784c51e6400893/pkg/auth/service.go#L1047